packages/slot/src/slot-manager.js

```diff
@@ -210,12 +239,12 @@ class SlotManager {
         end: result.utterance.length - 1,
         len: result.utterance.length,
       });
-    }
-    for (let i = 0, l = result.entities.length; i < l; i += 1) {
-      delete mandatorySlots[result.entities[i].entity];
+      delete mandatorySlots[context.slotFill.currentSlot];
     }
     keys = Object.keys(mandatorySlots);
     if (!keys || keys.length === 0) {
+      // All mandatory slots are filled, so we are done. No further questions needed
+      delete result.srcAnswer;
```

Comment:
As far as I know, the purpose of srcAnswer was to keep track of the raw answer without being processed at all.

Example:
answer:  "Ok let's play some rock"
srcAnswer:  "Ok let's play some {{style}}"

Reply:
In this case we only talk about slot filling quertions and not "In general".

result.srcAnswer is initialized with the last slotfill answer aka "question" from the last run (line 205 this file). And was in the past the also preserved when the latest user input resolved all open manatory slots. This then had effects and the same question as last time was potentially asked again by slot filling algorithm.

This then had the effect that because srcAnswer was still existing, but not relevant anymore.

When digging through code I can not find other places where srcAnswer is used than for slotfilling.

In fact your idea makes a lot of sense and could be added. The past code have set srcAnswer in slot filling but Never replaced contained entities at all ... I added that :) So I change that noew here to have srcAnswer as the raw text. Good idea

Reply:
PS: if you would like to introduce srcAnswer as field in general also for the normal answers, this can be done with changes in nlg and nlp. Should be straight forward.

Reply:
we can leave it like this by now but have it on mind for next updates, thanks!